src/gpu/intel/softmax/simple.hpp

```diff
@@ -73,6 +77,7 @@ struct simple_fwd_t : public primitive_t {
             VDISPATCH_SOFTMAX(attr()->has_default_values(skip_mask_t::scales
                                       | skip_mask_t::post_ops),
                     VERBOSE_UNSUPPORTED_ATTR);
+            VDISPATCH_SOFTMAX(is_double_blk, VERBOSE_UNSUPPORTED_TAG);
```

Comment:
Since this appears to be an actual use case, why don't we just modify the softmax implementation to use [OFF_MD](https://github.com/uxlfoundation/oneDNN/blob/f47f4e9261ec7038745849efb9418e6446f335db/src/gpu/intel/include/types.h#L945-L946) for offset calculation? I don't see any reason softmax should have custom (and apparently incorrect) logic for offset calculations.

Reply:
The idea would be to just remove those macros. The end result would effectively be to use calls like in [ref_prelu](https://github.com/uxlfoundation/oneDNN/blob/88617595be725e313fa07df5e70633f808cfa248/src/gpu/intel/prelu/ref.cl#L33-L35). In order to use those macros, all that should be required is to add a call to [def_memory_desc_info](https://github.com/uxlfoundation/oneDNN/blob/88617595be725e313fa07df5e70633f808cfa248/src/gpu/intel/prelu/ref.cpp#L89-L91) on the relevant buffers during kernel_ctx initialization.

> Does OFF_MD cater for double blocking?

Yes, it should support double blocking, which is largely why the [defintion](https://github.com/uxlfoundation/oneDNN/blob/88617595be725e313fa07df5e70633f808cfa248/src/gpu/intel/include/types.h#L913-L943) is so long.